docs/plots/backends.rst

```diff
@@ -0,0 +1,389 @@
+implement some of these as a no-op, e.g. because the underlying plotting library
+does not support 2D data. In that case, the backend would typically issue a
+warning.
+
+The plotting functions are not separated by "how things look on paper" (thus "plot" is
+a long method that is responsible for points, lines, and error bars), but
+by "what is the input data type":
+
+- `~sherpa.plot.backend.BaseBackend.plot` (for scatter plots with marker style
+  set, for line plots with line style set, and for error bars with ``xerr`` or
+  ``yerr`` set to `True`); accepts (x, y) data with optional error bars in each
+  dimension. Data can be scalar (for a single marker), or array-like.
+  Note that ``x`` and ``y`` can also be `None`, which should create an empty plot.
+- `~sherpa.plot.backend.BaseBackend.histo` (similar to plot, but with
+  "histogram-style" lines); accepts (xlo, xhi, y) data with optional xerr, yerr.
+- `~sherpa.plot.backend.BaseBackend.contour` for (x0, x1, z) data.
+
+If called with empty data, a plotting function shall at a minimum create an
+empty plot.
+
+Annotations
+-----------
+
+Backends should also implement the following annotation functions. They do not
+depend on the data plotted, but just annotate the plot, e.g. a
+`~sherpa.plot.RatioPlot` shows the ratio between data and model and can use an
+annotation to mark the ``ratio=1`` line.
+
+- `~sherpa.plot.backend.BaseBackend.hline` (horizontal across the entire axes)
+- `~sherpa.plot.backend.BaseBackend.vline` (vertical across the entire axes)
+
+Other annotations (e.g. text labels) might be added to the API in the future.
+For this reason new backends should inherit from
+`~sherpa.plot.backend.BasicBackend`. Any function added to the API will be
+implemented in `~sherpa.plot.backend.BasicBackend` as a no-op with a
+warning to the user like "Feature XYZ is not available in your backend". That
+way, all Sherpa plots can immediately make use of newly added functions without
+breaking existing plotting backends; the worst that happens is that not all
+annotation will be visible in every backend.
+
+Return values
+-------------
+
+Sherpa does not expect a specific return argument from any plotting function,
+but they are allowed to have return values if that is helpful for their internal
+implementation, e.g. in the `~sherpa.plot.pylab_backend.PylabBackend` backend, plotting a line might return a
+line object so that error bars plotted later can use ``line.color`` to match the
+color of that line.
+
+Creating plots and panels, clearing and overplotting
+----------------------------------------------------
+.. todo::
+   Add details
+
+Each of the plotting functions above accepts
+the following arguments: title, xlabel, ylabel, xlog, ylog, overplot, clearwindow
+
+Multi-panels plot can be set with ``set_subplot`` and ``set_jointplot``.
+See `sherpa.plot.backends.BaseBackend` for a listing of the calling signature.
```

Comment:
A possibility:

```suggestion
Multi-panels plot can be set with `~sherpa.plot.backends.BaseBackend.set_subplot``
and `~sherpa.plot.backends.BaseBacken.set_jointplot` (from `sherpa.plot.backends.BaseBackend`).
```

An advantage is that the HTML viewer can easily follow the link, and I kept the link to the class since we want it easy for the user to follow what class to use as a base.

Of course, given #1868 we may be reworking this section, but it's not obvious to me we'll get that done for 4.16.0.